Kaggle and Hugging Face integration post (#2855)
huggingface/blog: kaggle-hugging-face-integration.md

---
title: "Kaggle and Hugging Face Partner & Integrate to Improve Model Access" 
thumbnail: /blog/assets/kaggle-hugging-face-integration/thumbnail.png
authors:

> - user: roseberryv
> - user: megrisdal

Reviewer: ```suggestion
- user: roseberryv
  guest: true
  org: kaggle
- user: megrisdal
  guest: true
  org: kaggle
```

This will show a nice org logo below your names.

- user: julien-c
- user: pcuenq
- user: reach-vb
---

Kaggle and Hugging Face users are part of one AI community. That’s why we’re excited to announce our plans to bring our platforms and communities closer to better serve AI developers everywhere.

Beginning today, Kaggle is launching an integration that enhances visibility and discoverability for Hugging Face models directly on Kaggle. 

## How to get started

You can navigate from Hugging Face models to Kaggle and vice versa. Start by visiting a Hugging Face model page like [Qwen/Qwen3-1.7B](https://huggingface.co/Qwen/Qwen3-1.7B). To use it in a Kaggle Notebook, you can click on “Use this model” and select “Kaggle” to open up a Kaggle notebook with a pre-populated code snippet to load the model. You can do the same from a Hugging Face model page on Kaggle by clicking the “Code” button.

> ![Creating a new notebook using a Hugging Face model on Kaggle](assets/kaggle-hugging-face-integration/new-notebook.gif)

Reviewer 2: is it possible for you to open a PR here: https://huggingface.co/datasets/huggingface/documentation-images/tree/main (you can make a folder there and then link those from the HF repo to here)

Author: Certainly, should I use the article slug as a folder or something shorter such as `kaggle`?

Reviewer: yes, article slug is good!

Reviewer 2: `kaggle-integration` is okay imo

Author: I opened a PR: https://huggingface.co/datasets/huggingface/documentation-images/discussions/485

I will update this PR to refer to the images in `documentation-images`.

## What’s next

We’re actively working on a solution to seamlessly use Hugging Face models in Kaggle competitions that require offline notebook submissions. While this will take a few more months to complete, we believe the wait will be worth it. 

You can read Kaggle’s position on [“AI Competitions as the gold standard for empirical rigor for GenAI evaluation”](https://huggingface.co/papers/2505.00612) to understand why it’s so important for us to get this part of the integration right! But tl;dr – Kaggle is highly sensitive to data leakage and its impact on model contamination.  Our goal is to design this integration to preserve the integrity of our competitions and their vital role in the industry, while enabling seamless access for Kaggle competitors to build with the best models from Hugging Face!

> We’d love to hear your feedback in the meantime - [share your thoughts and ideas here](https://huggingface.co/spaces/kaggle/hf-integration-feedback/discussions)!

Reviewer 2: ```suggestion
We’d love to hear your feedback in the meantime - [share your thoughts and ideas here](https://huggingface.co/spaces/kaggle/hf-integration-feedback/discussions/1)!
```

---
title: "Kaggle and Hugging Face Partner & Integrate to Improve Model Access" 
thumbnail: /blog/assets/kaggle-hugging-face-integration/thumbnail.png
authors:
- user: roseberryv
  guest: true
  org: kaggle
- user: megrisdal
  guest: true
  org: kaggle
- user: julien-c
- user: pcuenq
- user: reach-vb
---

Kaggle and Hugging Face users are part of one AI community. That’s why we’re excited to announce our plans to bring our platforms and communities closer to better serve AI developers everywhere.

Beginning today, Kaggle is launching an integration that enhances visibility and discoverability for Hugging Face models directly on Kaggle. 

## How to get started

You can navigate from Hugging Face models to Kaggle and vice versa. Start by visiting a Hugging Face model page like [Qwen/Qwen3-1.7B](https://huggingface.co/Qwen/Qwen3-1.7B). To use it in a Kaggle Notebook, you can click on “Use this model” and select “Kaggle” to open up a Kaggle notebook with a pre-populated code snippet to load the model. You can do the same from a Hugging Face model page on Kaggle by clicking the “Code” button.

> ![Creating a new notebook using a Hugging Face model on Kaggle](assets/kaggle-hugging-face-integration/new-notebook.gif)

Reviewer: ```suggestion
![Creating a new notebook using a Hugging Face model on Kaggle](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/kaggle-integration/new-notebook.gif)
```

When you run a notebook on Kaggle that references a model hosted on Hugging Face Hub, we will automatically generate a Hugging Face model page if one doesn’t exist already. You don’t need to make any special changes to your code.  Additionally, when you make your notebook public, it will automatically show on the “Code” tab of the Kaggle model page.

Discover Hugging Face models and explore all the community examples in public notebooks in one place on Kaggle at [https://www.kaggle.com/models](https://www.kaggle.com/models). As more Hugging Face models are used on Kaggle, the number of models and associated code examples you can explore for inspiration will grow.

> ![Browsing Hugging Face models on Kaggle](assets/kaggle-hugging-face-integration/browsing-hugging-face-models-on-kaggle.gif)

Reviewer: ```suggestion
![Browsing Hugging Face models on Kaggle](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/kaggle-integration/browsing-hugging-face-models-on-kaggle.gif)
```
